isisd/isis_sr.h

```diff
@@ -57,11 +57,25 @@
 #define SRLB_UPPER_BOUND               15999
 
 /* Segment Routing Data Base (SRDB) RB-Tree structure */
-PREDECL_RBTREE_UNIQ(srdb_node)
-PREDECL_RBTREE_UNIQ(srdb_node_prefix)
-PREDECL_RBTREE_UNIQ(srdb_area_prefix)
 PREDECL_RBTREE_UNIQ(srdb_prefix_cfg)
 
+/*
+ * Segment Routing Prefix-SID information.
+ *
+ * This structure is intended to be embedded inside other structures that
+ * might or might not contain Prefix-SID information.
+ */
+struct isis_sr_psid_info {
+	/* Prefix-SID Sub-TLV information. */
+	struct isis_prefix_sid sid;
+
+	/* Resolved input/output label. */
+	mpls_label_t label;
+
+	/* Indicates whether the Prefix-SID is present or not. */
+	bool present;
```

Comment:
MPLS_INVALID_LABEL value for the label could also server to indicate if the Prefix-SID is present or not. May be present is superfluous in this case

Reply:
Forget this remark.